### ISSUE TITLE
SpeechAutoEncoder Model and HifiGan Checkpoints

### ISSUE THREAD

**Issue by HLSUD** [CCTN-BCI/Neural2Speech#1]
Your work is really impressive. In your code, I could not find the SpeechAutoEncoder class. Could you please provide the code for this part and the weights? Many thanks.

**Comment by CCTN-BCI** [comment]
Thank you for your appreciation. We did not provide the class because the arechitecture on both components are available online. If you need the exact codes and parameters, please let me know how I should send these files to you.

It seems that you can speak Chinese because of your name. Can I upload the files to BaiduNetDisk? Maybe it is more accessible for both of us.

**Comment by HLSUD** [comment]
Thank you for your prompt reply. I am Chinese and BaiduNetDisk works for me. Could you please share a link of the code and weights? Thanks

**Comment by CCTN-BCI** [comment]
link:
https://pan.baidu.com/s/1W8IZQbvHAFaUTn_9NQAwCw?pwd=ohf4 
password:
ohf4

> Thank you for your prompt reply. I am Chinese and BaiduNetDisk works for me. Could you please share a link of the code and weights? Thanks

**Comment by CCTN-BCI** [comment]
> Thank you for your prompt reply. I am Chinese and BaiduNetDisk works for me. Could you please share a link of the code and weights? Thanks

link:
https://pan.baidu.com/s/1W8IZQbvHAFaUTn_9NQAwCw?pwd=ohf4
password:
ohf4

**Comment by HLSUD** [comment]
Thank you for the link. I have a question about training HiFi Gan. Did you freeze the wav2vec encoder when you train the hifi Gan?

**Comment by CCTN-BCI** [comment]
> Thank you for the link. I have a question about training HiFi Gan. Did you freeze the wav2vec encoder when you train the hifi Gan?

Yes.  It is the same as DatatoVecAudio in Huggingface. 
The annotations shows that we used the same model archetecture as wav2vec2.0 (you can find the source code at the python packages installed in your computer).

**Comment by aparna9625** [comment]
Hi, could you also share with me the model checkpoint & the SAE class? Would it possible to add it to the repo?

**Comment by CCTN-BCI** [comment]
> Hi, could you also share with me the model checkpoint & the SAE class? Would it possible to add it to the repo?

I can share the model checkpoint & the SAE class. It is too large to add it to the repo but I can send to you in other ways.  Is upgrading the BaiduNetDisk Link a feasible way for you?

**Comment by aparna9625** [comment]
Can upload it to Google Drive or Dropbox?

**Comment by CCTN-BCI** [comment]
> Can upload it to Google Drive or Dropbox?

Certainly. The files are available in the link below. Demo Python codes and demo speech waveform are also attached for you.

https://www.dropbox.com/scl/fi/qverx4jo6wj7o9mieit34/pre-trained_SAE.zip?rlkey=77e3z062iuiuu994v6jq8piqa&dl=0

**Comment by hawkingwoo** [comment]
> > Thank you for your prompt reply. I am Chinese and BaiduNetDisk works for me. Could you please share a link of the code and weights? Thanks
> 
> link: https://pan.baidu.com/s/1W8IZQbvHAFaUTn_9NQAwCw?pwd=ohf4 password: ohf4

The link seems be expired. Could you share it again?

**Comment by CCTN-BCI** [comment]
> > > Thank you for your prompt reply. I am Chinese and BaiduNetDisk works for me. Could you please share a link of the code and weights? Thanks
> > 
> > 
> > link: https://pan.baidu.com/s/1W8IZQbvHAFaUTn_9NQAwCw?pwd=ohf4 password: ohf4
> 
> The link seems be expired. Could you share it again?

You can see the dropbox link mentioned above. You know,  BaiduNetDisk sometimes cannot provide you a high download speed. https://www.dropbox.com/scl/fi/qverx4jo6wj7o9mieit34/pre-trained_SAE.zip?rlkey=77e3z062iuiuu994v6jq8piqa&dl=0

**Comment by hawkingwoo** [comment]
> > > > Thank you for your prompt reply. I am Chinese and BaiduNetDisk works for me. Could you please share a link of the code and weights? Thanks
> > > 
> > > 
> > > link: https://pan.baidu.com/s/1W8IZQbvHAFaUTn_9NQAwCw?pwd=ohf4 password: ohf4
> > 
> > 
> > The link seems be expired. Could you share it again?
> 
> You can see the dropbox link mentioned above. You know, BaiduNetDisk sometimes cannot provide you a high download speed. https://www.dropbox.com/scl/fi/qverx4jo6wj7o9mieit34/pre-trained_SAE.zip?rlkey=77e3z062iuiuu994v6jq8piqa&dl=0

Thanks! Yes, higher speed than baidu.
